accountsdb-plugin-manager/src/accountsdb_plugin_service.rs

```diff
@@ -0,0 +1,131 @@
+use {
+    crate::{
+        accounts_update_notifier::{AccountsSelector, AccountsUpdateNotifier},
+        accountsdb_plugin_manager::AccountsDbPluginManager,
+        confirmed_slots_observer::SlotConfirmationObserver,
+    },
+    crossbeam_channel::Receiver,
+    log::*,
+    serde_json,
+    solana_runtime::bank_forks::BankForks,
+    solana_sdk::clock::Slot,
+    std::{
+        fs::File,
+        io::Read,
+        path::Path,
+        sync::{Arc, RwLock},
+        thread,
+    },
+};
+
+/// The service managing the AccountsDb plugin workflow.
+pub struct AccountsDbPluginService {
+    confirmed_slots_observer: SlotConfirmationObserver,
+    plugin_manager: Arc<RwLock<AccountsDbPluginManager>>,
+}
+
+impl AccountsDbPluginService {
+    /// Creates and returns the AccountsDbPluginService.
+    /// # Arguments
+    /// * `confirmed_bank_receiver` - The receiver for confirmed bank notification
+    /// * `bank_forks` - The bank forks.
+    /// * `accountsdb_plugin_config_file` - The config file path for the plugin. The
+    ///    config file controls what accounts to stream and the plugin responsible
+    ///    for transporting the data to external data stores. It is defined in JSON format.
+    ///    The `libpath` field should be pointed to the full path of the dynamic shared library
+    ///    (.so file) to be loaded. The shared library must implement the `AccountsDbPlugin`
+    ///    trait. And the shared library shall export a `C` function `_create_plugin` which
+    ///    shall create the implementation of `AccountsDbPlugin` and returns to the caller.
+    ///    The `accounts_selector` section allows the user to controls accounts selections.
+    ///    accounts_selector = {
+    ///         accounts = \['pubkey-1', 'pubkey-2', ..., 'pubkey-n'\],
+    ///    }
+    ///    or:
+    ///    accounts_selector = {
+    ///         owners = \['pubkey-1', 'pubkey-2', ..., 'pubkey-m'\]
+    ///    }
+    ///    When accounts and owners are specified, the program only filters only using accounts
+    ///    as it is more selective. And owners field is ignored. When only owners is specified,
```

Comment:
I had some editing suggestions here, but first:
Is this true that the owners field is ignored? It looks to me that due to `AccountsSelector::is_account_selected()`, the results will be the union of accounts specified in `accounts` and those owned by `owners`.

Reply:
Corrected.